nsxt/resource_nsxt_lb_http_monitor.go

```diff
@@ -0,0 +1,209 @@
+		return fmt.Errorf("Error during LbHttpMonitor read: %v", err)
+	}
+
+	d.Set("revision", lbHTTPMonitor.Revision)
+	d.Set("description", lbHTTPMonitor.Description)
+	d.Set("display_name", lbHTTPMonitor.DisplayName)
+	setTagsInSchema(d, lbHTTPMonitor.Tags)
+	d.Set("fall_count", lbHTTPMonitor.FallCount)
+	d.Set("interval", lbHTTPMonitor.Interval)
+	d.Set("monitor_port", lbHTTPMonitor.MonitorPort)
+	d.Set("rise_count", lbHTTPMonitor.RiseCount)
+	d.Set("timeout", lbHTTPMonitor.Timeout)
+	d.Set("request_body", lbHTTPMonitor.RequestBody)
+	setLbHTTPHeaderInSchema(d, "request_header", lbHTTPMonitor.RequestHeaders)
+	d.Set("request_method", lbHTTPMonitor.RequestMethod)
+	d.Set("request_url", lbHTTPMonitor.RequestUrl)
+	d.Set("request_version", lbHTTPMonitor.RequestVersion)
+	d.Set("response_body", lbHTTPMonitor.ResponseBody)
+	d.Set("response_status_codes", int32List2Interface(lbHTTPMonitor.ResponseStatusCodes))
+
+	return nil
+}
+
+func resourceNsxtLbHTTPMonitorUpdate(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(*api.APIClient)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining logical object id")
+	}
+
+	revision := int32(d.Get("revision").(int))
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	tags := getTagsFromSchema(d)
+	fallCount := int64(d.Get("fall_count").(int))
+	interval := int64(d.Get("interval").(int))
+	monitorPort := d.Get("monitor_port").(string)
+	riseCount := int64(d.Get("rise_count").(int))
+	timeout := int64(d.Get("timeout").(int))
+	requestBody := d.Get("request_body").(string)
+	requestHeaders := getLbHTTPHeaderFromSchema(d, "request_header")
+	requestMethod := d.Get("request_method").(string)
+	requestURL := d.Get("request_url").(string)
+	requestVersion := d.Get("request_version").(string)
+	responseBody := d.Get("response_body").(string)
+	responseStatusCodes := interface2Int32List(d.Get("response_status_codes").([]interface{}))
+	lbHTTPMonitor := loadbalancer.LbHttpMonitor{
+		Revision:            revision,
+		Description:         description,
+		DisplayName:         displayName,
+		Tags:                tags,
+		FallCount:           fallCount,
+		Interval:            interval,
+		MonitorPort:         monitorPort,
+		RiseCount:           riseCount,
+		Timeout:             timeout,
+		RequestBody:         requestBody,
+		RequestHeaders:      requestHeaders,
+		RequestMethod:       requestMethod,
+		RequestUrl:          requestURL,
+		RequestVersion:      requestVersion,
+		ResponseBody:        responseBody,
+		ResponseStatusCodes: responseStatusCodes,
+	}
+
+	lbHTTPMonitor, resp, err := nsxClient.ServicesApi.UpdateLoadBalancerHttpMonitor(nsxClient.Context, id, lbHTTPMonitor)
+
+	if err != nil || resp.StatusCode == http.StatusNotFound {
+		return fmt.Errorf("Error during LbHttpMonitor update: %v", err)
+	}
+
+	return resourceNsxtLbHTTPMonitorRead(d, m)
+}
+
+func resourceNsxtLbHTTPMonitorDelete(d *schema.ResourceData, m interface{}) error {
```

Comment:
Seems like the delete function is the same for all monitors. Maybe it can be in lb_utils too.